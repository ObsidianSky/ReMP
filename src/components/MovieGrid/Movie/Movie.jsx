import React from 'react';
import PropTypes from 'prop-types';
import './Movie.scss'

export default class Movie extends React.Component {
    render() {
        const { img, title, year, genre } = this.props.movie;

        return (
            <div className="movie">
                <img className="movie__image" src={img}></img>
                <div className="movie__info">
                    <div className="movie__info-row">
                        <div className="movie__title">{title}</div>
                        <div className="movie__year">{year}</div>
                    </div>
                    <div className="movie__info-row">
                        <div className="movie__genre">{genre}</div>
                    </div>
                </div>
            </div>
        )
    }
}

Movie.propTypes = {
    movie: PropTypes.shape({
        img: PropTypes.string,
        title: PropTypes.string,
        year: PropTypes.number,
        genre: PropTypes.string
    })
};