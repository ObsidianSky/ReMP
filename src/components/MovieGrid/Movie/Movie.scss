@import '../../../assets/styles/variables';

.movie {
  display: flex;
  flex-direction: column;
  height: 100%;

  &__image-link {
    flex: 1 0 auto;
    margin-bottom: 15px;
    cursor: pointer;
    display: flex;
    align-items: center;
  }

  &__image {
    width: 100%;
  }

  &__info-row {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 8px;
  }

  &__title {
    color: $text-black;
    font-size: 20px;
    font-weight: 600;
    margin-right: 10px;
    text-transform: uppercase;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    cursor: pointer;
  }

  &__year {
    color: $text-black;
    font-size: 16px;
    padding: 2px 6px;
    border: 1px solid $text-dark;
    font-weight: 600;
  }

  &__genre {
    font-size: 14px;
    color: $text;
  }

  &__rating {
    color: $green;
    font-size: 11px;
    width: 22px;
    height: 22px;
    line-height: 20px;
    text-align: center;
    border-radius: 50%;
    border: 1px solid $green;
  }
}