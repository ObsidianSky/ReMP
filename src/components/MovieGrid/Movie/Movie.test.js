import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Movie from './Movie';

describe('Movie', () => {
    let props;
    let onMovieClick;

    beforeEach(() => {
        onMovieClick = jest.fn();
        props = {
            img: 'img',
            title: 'title',
            year: 'year',
            genre: 'genre',
            rating: 'rating',
            onMovieClick
        }
    });

    test('should renders correctly', () => {
        const wrapper = shallow(<Movie { ...props } />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should call provided callback on image click', () => {
        const wrapper  = shallow(<Movie { ...props } />);

        wrapper .find('.movie__image-link').simulate('click');

        expect(onMovieClick).toHaveBeenCalled();
    });

    test('should call provided callback on title click', () => {
        const wrapper  = shallow(<Movie { ...props } />);

        wrapper .find('.movie__title').simulate('click');

        expect(onMovieClick).toHaveBeenCalled();
    });
});

