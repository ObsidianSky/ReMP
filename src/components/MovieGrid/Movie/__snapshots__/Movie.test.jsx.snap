// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should renders correctly 1`] = `
<div
  className="movie"
>
  <div
    className="movie__image-link"
    onClick={[Function]}
  >
    <img
      className="movie__image"
      src="img"
    />
  </div>
  <div
    className="movie__info"
  >
    <div
      className="movie__info-row"
    >
      <div
        className="movie__title"
        onClick={[Function]}
        title="title"
      >
        title
      </div>
      <div
        className="movie__year"
      >
        year
      </div>
    </div>
    <div
      className="movie__info-row"
    >
      <div
        className="movie__genre"
      >
        genre
      </div>
      <div
        className="movie__rating"
      >
        rating
      </div>
    </div>
  </div>
</div>
`;
